frame_utils_test.go

```diff
@@ -93,3 +107,44 @@ func (p *RecordingFramePool) CheckEmpty() (int, string) {
 	}
 	return len(p.allocations), strings.Join(badCalls, "\n")
 }
+
+// NewProtectMemFramePool creates a frame pool that ensures that released frames
+// are not reused by removing all access to a frame once it's been released.
+func NewProtectMemFramePool() FramePool {
+	return &ProtectMemFramePool{
+		allocations: make(map[*Frame]protectMemAllocs),
+	}
+}
+func (p *ProtectMemFramePool) Get() *Frame {
+	frameAlloc := protectmem.Allocate(unsafe.Sizeof(Frame{}))
+	f := (*Frame)(frameAlloc.Ptr())
+
+	bufferAlloc := protectmem.AllocateSlice(&f.buffer, MaxFramePayloadSize)
+	f.Payload = f.buffer[:MaxFramePayloadSize]
+	f.buffer = f.buffer[:MaxFramePayloadSize]
+	f.Payload = f.buffer[FrameHeaderSize:]
+	f.headerBuffer = f.buffer[:FrameHeaderSize]
```

Comment:
Looks like there's a copy-paste bug here - `frame.Payload` is being set to two different things here.